src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/dateRange.jsx

```diff
@@ -74,29 +68,20 @@ class DateRange extends React.Component {
     maxPickableDays: MAX_PICKABLE_DAYS,
   };
 
-  static getTimeStringFromDate = (date, utc) => {
-    return getFormattedDate(date, 'HH:mm', {local: !utc});
+  static getTimeStringFromDate = date => {
+    return moment(date)
+      .local()
+      .format('HH:mm');
   };
 
   handleSelectDateRange = ({selection}) => {
-    const {utc, onChange} = this.props;
+    const {onChange} = this.props;
     const {startDate, endDate} = selection;
 
-    let start = startDate;
-    let end = endDate;
-
-    if (start) {
-      start = setDateToTime(start, DEFAULT_DAY_START_TIME, {local: !utc});
-    }
-
-    if (end) {
-      end = setDateToTime(end, DEFAULT_DAY_END_TIME, {
-        local: !utc,
-      });
-    }
+    const end = endDate ? getEndOfDay(endDate) : endDate;
```

Comment:
When selecting a date range, the start/end times are at 00:00, to make end date inclusive, need to get the end of the day.